lib/jyson.js

```diff
@@ -72,7 +86,7 @@ module.exports.buildTemplateFunction = (templateObject, buildTemplateFunctionOpt
     return objects.map(object => {
       const json = _.cloneDeep(templateObject);
 
-      return fillKeys(json, object, buildTemplateFunctionOpts.undefinedValue, opts);
+      return fillKeys(json, object, buildTemplateFunctionOpts.undefinedValue, Object.assign(opts, {arrayIndex: {'$': 0}}));
```

Comment:
What's the difference between `json` and `object`?

Reply:
`json` is the json we are generating
`object` is the object we are generating the json from

Open to better names / documentation.